commit 2927113c0bd50cf2eaf2c7f63ed83b9a645e7054
Author: q66 <q66@chimera-linux.org>
Date:   Tue Apr 16 18:46:10 2024 +0200

    disable memchr
    
    since C11 it's well defined if the character is found within
    the array even if the array is smaller than the given size

diff --git a/include/string.h b/include/string.h
index db2e6c3..6f90e6f 100644
--- a/include/string.h
+++ b/include/string.h
@@ -30,7 +30,6 @@ extern "C" {
 #endif
 
 #undef memcpy
-#undef memchr
 #undef memmove
 #undef memset
 #undef strcat
@@ -111,6 +110,7 @@ __warning_if(__c != 0 && __n == 0, "'memset' will set `0` bytes; did you invert
 #endif
 }
 
+#if 0
 __fh_access(read_only, 1, 3)
 #if __has_builtin(__builtin_memchr)
 __diagnose_as_builtin(__builtin_memchr, 1, 2, 3)
@@ -132,6 +132,7 @@ _FORTIFY_FN(memchr) void *memchr(const void * _FORTIFY_POS0 __d, int __c, size_t
 	return __builtin_memchr(__d, __c, __n);
 #endif
 }
+#endif
 
 #if 0
 __fh_access(read_only, 1, 2)
diff --git a/tests/Makefile b/tests/Makefile
index b1223ea..3213319 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -46,9 +46,6 @@ RUNTIME_TARGETS= \
 	test_mbsrtowcs_static        \
 	test_mbstowcs_dynamic        \
 	test_mbstowcs_static         \
-	test_memchr_dynamic_read     \
-	test_memchr_static_read      \
-	test_memchr_null             \
 	test_memcpy_dynamic_read     \
 	test_memcpy_dynamic_write    \
 	test_memcpy_overwrite_over   \
